Yank/repex.py

```diff
@@ -259,13 +259,15 @@ def write_thermodynamic_states(self, thermodynamic_states, unsampled_states):
                 # Write state as dictionary.
                 self.write_dict('{}/state{}'.format(state_type, state_id), serialized_state)
 
-    def read_sampler_states(self, iteration):
+    def read_sampler_states(self, iteration=slice(None)):
```

Comment:
Im fine with converting it back into an `arg` and allowing for the `slice(None)` on demand from the users. When I added those in, I kind of blindly just went through, only realizing after some of them would not work. So lets revert that.